posthog/api/organization_feature_flag.py

```diff
@@ -56,11 +57,19 @@ def copy_flags(self, request, *args, **kwargs):
         if not feature_flag_key or not from_project or not target_project_ids:
             return Response({"error": "Missing required fields"}, status=status.HTTP_400_BAD_REQUEST)
 
+        # Fetch the flag to copy
         try:
             flag_to_copy = FeatureFlag.objects.get(key=feature_flag_key, team_id=from_project)
         except FeatureFlag.DoesNotExist:
             return Response({"error": "Feature flag to copy does not exist."}, status=status.HTTP_400_BAD_REQUEST)
 
+        # Check if the user is allowed to edit the flag
+        can_edit_feature_flag = CanEditFeatureFlag()
```

Comment:
This is good! Happy to merge 👍 

Although, I'm not 100% sure what happens in this case - I think the API will fail, just not yet sure how it's represented in the UI yet:

1. By default, everyone has view permissions only.
2. On the current flag, user has permission to edit flags because, say, they created the flag here. [because they were in a role that allows editing]
3. They are no longer in a role that allows editing flags.
4. Now, trying to copy to another project should fail, and I think it will because of this check, but not sure what will happen on the UI.

Can you check this case?

Reply:
> ...say, they created the flag here. [because they were in a role that allows editing]
Actually, any user is currently allowed to create a new flag, regardless of the Edit permissions. Was this not intended?

Also, the user that created the flag is always allowed to edit it, regardless of the Edit permissions.

I tested the case when they are temporarily assigned Edit rights to the flag created by someone else, and then lose the rights - the UI works correctly here.

One more case - say they copy a flag of someone else, and then lose Edit rights. They will still have access to the new flag in the destination project, because when copying, they become the creator of the new flag. Perhaps we want to set _created_by_ of the new flag to the owner of the original flag?

Reply:
"You may be the ruler in prod, but I control dev" 😈 

😂  

if new cases around this come up later, I think reasonable to address them as they come, given that permissions isn't often used, and we're at the edge-edge of using permissions + copying flags. And we don't have a good sense of how these will be used together / what users will want the default behaviour to be here

Reply:
Makes sense!